lib/Downloader.js

```diff
@@ -34,70 +35,51 @@ const downloadURLs = {
   win64: '%s/chromium-browser-snapshots/Win_x64/%d/chrome-win32.zip',
 };
 
-// Project root will be different for node6-transpiled code.
-const PROJECT_ROOT = fs.existsSync(path.join(__dirname, '..', 'package.json')) ? path.join(__dirname, '..') : path.join(__dirname, '..', '..');
+const readdirAsync = helper.promisify(fs.readdir.bind(fs));
+const mkdirAsync = helper.promisify(fs.mkdir.bind(fs));
+const unlinkAsync = helper.promisify(fs.unlink.bind(fs));
 
-class Downloader {
-  /**
-   * @param {string} downloadsFolder
-   */
-  constructor(downloadsFolder) {
-    this._downloadsFolder = downloadsFolder;
-    this._downloadHost = DEFAULT_DOWNLOAD_HOST;
-  }
-
-  /**
-   * @return {string}
-   */
-  static defaultRevision() {
-    return require(path.join(PROJECT_ROOT, 'package.json')).puppeteer.chromium_revision;
-  }
-
-  /**
-   * @return {!Downloader}
-   */
-  static createDefault() {
-    const downloadsFolder = path.join(PROJECT_ROOT, '.local-chromium');
-    return new Downloader(downloadsFolder);
-  }
-
-  /**
-   * @param {string} downloadHost
-   */
-  setDownloadHost(downloadHost) {
-    this._downloadHost = downloadHost.replace(/\/+$/, '');
-  }
+function existsAsync(filePath) {
+  let fulfill = null;
+  const promise = new Promise(x => fulfill = x);
+  fs.access(filePath, err => fulfill(!err));
+  return promise;
+}
 
+class Downloader {
   /**
-   * @return {!Array<string>}
+   * @param {!Downloader.Options=} options
    */
-  supportedPlatforms() {
-    return Object.keys(downloadURLs);
+  constructor(options = {}) {
+    this._downloadsFolder = options.path || path.join(helper.projectRoot(), '.local-chromium');
+    this._downloadHost = options.host || DEFAULT_DOWNLOAD_HOST;
+    this._platform = options.platform || '';
+    if (!this._platform) {
+      const platform = os.platform();
+      if (platform === 'darwin')
+        this._platform = 'mac';
+      else if (platform === 'linux')
+        this._platform = 'linux';
+      else if (platform === 'win32')
+        this._platform = os.arch() === 'x64' ? 'win64' : 'win32';
```

Comment:
console.assert(this._platform, 'Unsupported platform: ' + os.platform());

Reply:
Done.